### ISSUE TITLE
NullPointerException when calling DBMS_AQ.dequeue

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13827]
A regression has been introduced in jOOQ 3.17:

```
java.lang.NullPointerException
at org.jooq.impl.AbstractExecuteScope.<init>(AbstractExecuteScope.java:53)
at org.jooq.impl.DefaultBindingGetSQLInputContext.<init>(DefaultBindingGetSQLInputContext.java:55)
at org.jooq.impl.AbstractQualifiedRecord.readSQL(AbstractQualifiedRecord.java:127)
at oracle.sql.STRUCT.toClass(STRUCT.java:612)
at oracle.sql.STRUCT.toJdbc(STRUCT.java:563)
at oracle.jdbc.driver.NamedTypeAccessor.getObject(NamedTypeAccessor.java:172)
at oracle.jdbc.driver.OracleCallableStatement.getObject(OracleCallableStatement.java:1956)
at oracle.jdbc.driver.OracleCallableStatementWrapper.getObject(OracleCallableStatementWrapper.java:647)
at org.jooq.util.oracle.OracleDSL$DBMS_AQ.lambda$dequeue$5(OracleDSL.java:1239)
at org.jooq.impl.DefaultDSLContext.connectionResult(DefaultDSLContext.java:671)
at org.jooq.util.oracle.OracleDSL$DBMS_AQ.dequeue(OracleDSL.java:1198)
```

Details to be investigated

**Comment by lukaseder** [comment]
The root cause is this one here: https://github.com/jOOQ/jOOQ/issues/13830

**Comment by lukaseder** [comment]
The root cause seems to be harder to fix (see https://github.com/jOOQ/jOOQ/issues/13830#issuecomment-1197946810), so I'll fix this NPE separately, first.

**Comment by lukaseder** [comment]
The regression itself was probably introduced with #12528

**Comment by lukaseder** [comment]
A quick test of #13830 shows that it would indeed solve this problem. I'll add an internal undocumented feature toggle to enable the fix for #13830 only for AQ usage.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.3 (https://github.com/jOOQ/jOOQ/issues/13831), and 3.16.9 (https://github.com/jOOQ/jOOQ/issues/13832)
